actors/evm/src/interpreter/precompiles/parameter.rs

```diff
@@ -18,212 +19,197 @@ impl From<GroupError> for PrecompileError {
     }
 }
 
-/// Pad out to len as needed, but does not slice output to len
-pub(super) fn right_pad<'a>(input: impl Into<Cow<'a, [u8]>>, len: usize) -> Cow<'a, [u8]> {
-    let mut input: Cow<[u8]> = input.into();
-    let input_len = input.len();
-    if len > input_len {
-        input.to_mut().resize(len, 0);
-    }
-    input
-}
-
-/// ensures top bits are zeroed
-pub fn assert_zero_bytes<const S: usize>(src: &[u8]) -> Result<(), PrecompileError> {
-    if src[..S] != [0u8; S] {
-        Err(PrecompileError::InvalidInput)
-    } else {
-        Ok(())
-    }
+pub(super) trait Parameter: Sized {
+    fn read(reader: &mut ParameterReader) -> Result<Self, PrecompileError>;
 }
 
-pub(super) struct Parameter<T>(pub T);
-
-impl<'a> TryFrom<&'a [u8; 64]> for Parameter<G1> {
-    type Error = PrecompileError;
```

Comment:
This dance got pretty confusing and meant that we couldn't read parameters with different sizes from the same input.